CONTRIBUTING.md

```diff
@@ -67,20 +65,52 @@ All pull requests must pass standard checks:
-in VCS. for pylint you may need to manually point it at `pylint.ini`
+Everything should be executable in your development environment, all necessary configuration files are provided
+in VCS. for pylint you may need to manually point at `pylint.ini`
 ## Hygiene
 
 **Please do not squash your commits prior to your PR.**  If this is necessary, it will be brought up during the review process.
 
-Your PR should have sensible commits and messages.
+Your PR should have sensible commits and messages.  Please see the **Pull Requests Naming & Commenting** section below.
 
-All new features are required to contain tests sufficient for newly added code to be evaluated without reducing coverage.
+New features are required to contain tests sufficient for newly added code to be evaluated without reducing coverage.
 
 After branching, you should immediately use `git push -u origin feature/my-awesome-feature` to make that the default upstream ref.
 
 Before opening a PR, rebase onto develop so your PR can be merged fast-forward only, without merge commits.
-The easiest way to do that is to make the main repo a remote using `git remote add upstream git@github.com:fuelrats/pipsqueak` and then running `git pull --rebase upstream develop`.
+The easiest way to do that is to make the main repository a remote using `git remote add upstream git@github.com:fuelrats/pipsqueak` and then running `git pull --rebase upstream develop`.
+
+## Naming & Commenting Standards
+### Pull Requests Naming
+If working from a registered issue, Include the issue name enclosed in brackets in the title of your Pull Request, ie `[SPARK-99] Update to CONTRIBUTING.md`
+
+Otherwise, use the type of request.
+``[Doc] Update TOS to revoke snickers delivery``
+``[Fix] Mechaclient.py should return None after valid on_message event``
+
+You may be asked to rename your Pull Request before peer review can begin if the name is not specific enough or not within standard.
+
+### Peer Review Tagging System
+During a peer review, reviewers will add a tag to the comment to help the author understand what type of action is required to receive an approving review.
+
+| Tag | Explanation |
+| --- | --- |
+[Blocking]|This issue must be fixed before approval.
+[Consensus]|Discussion is required.  Blocked until consensus has been reached by the author and reviewer.
+[Discussion]|Discussion is required, but this is not blocking an approval.
+[Incomplete]|Incomplete code and/or Pull Request.
+[Kudos]|The reviewer liked this, specifically. Go you.
+[Nitpick]|Same as suggestion.  Not blocking, and at authors discretion to change.
+[Style]|Breaking style guidelines, or other issue specifically with code style.
+[Scope]|Out of scope for your Pull Request.
+[Suggestion]|Suggested change.  Not blocking, and at authors discretion to change.
+[Testing]|Reviewer requests a usage case, or testing to validate. Not blocking.
+[Testing Needed]|Testing or usage case needed before approval. Blocking.
+
```

Comment:
[Suggestion] Add a [Questions] tag for things that don't look wrong, but may need a bit more clarification, or are just questions by that person specifically.

Reply:
It may just be me, but I think this is well covered by the [Discussion] tag.

Reply:
Discussion and questions could cover the same topics. I guess one leans towards an individual response and the other a collective response?

Reply:
For me, a Discussion is about how it **should** work, whereas a Question is about how it **does** work. They can indeed overlap.

Reply:
For me, I think there is little distinction between what necessitates a discussion and what necessitates a question. As it were, Questions often lead to a discussion as a response.

Thus, while not Discussions are results of questions, I do believe all questions result in discussions.
I think these two terms can be combined without loss in meaning.

Reply:
Is this blocking?  I dont see why we can't have two distinct separate meanings.  One indicates a discussion between the author and the reviewer, the other is a 'Full stop, this needs to be a group decision' tag.

Reply:
I wonder why we're attempting to define these as blocking/non-blocking. Instead of requiring separate tags, why not make a [Blocking] tag? This way the reviewer can block a merge regardless of other tags on the review.

Reply:
The idea is that some issues are blocking by the very nature of the issue, such as Error, but technically any issue could be blocking, dependent on the reviewer.

I'd like to keep the flexibility in review comments, because we need to make tags available for all reviewers without putting them in a box.

Reply:
My concept is any comment is blocking until addressed. If someone raises a question the Pr is blocked until it gets a response or is otherwise resolved. 

The only thing i can see that does not inherently block isa  kudos.

Reply:
That is certainly an way to review things, but it is not the intent of the proposal as discussed in IRC, nor is it the way we have historically reviewed pull requests.  Many times suggestions and nitpicks have been brought up that have explicitly been stated as not blocking.

Reply:
Historically it *has* been how pull requests have been treated, unless explicitly enumerated otherwise.

Reply:
At the very least, IMHO, suggestions should be acknowledged. at the very least, questions deserve a response (unless marked as resolved by the issuer due to PEBCAK etc)

Reply:
in light of that, can you detail the changes you'd make?  Are you suggesting we invert the system, and all tags are blocking unless we add a [Not Blocking] tag?

Reply:
as per conversations in IRC, i believe this conversation is resolved.